### PR TITLE
LSR-1137 Workflow list screenlet for Android

### PR DESCRIPTION
**Pull request by dgarciasarai**

### CODE REVIEW

File: android/build.gradle [hunk 1]
```diff
@@ -29,7 +29,7 @@ ext {
 
 	bintrayRepo = 'liferay-mobile'
 	bintrayOrg = ''
-//	    bintrayOrg = 'liferay'
+	//bintrayOrg = 'liferay'
```
Comment (victorg1991) [inline android/build.gradle]:
```suggestion
//          bintrayOrg = 'liferay'
```

File: android/library/src/main/java/com/liferay/mobile/screens/thingscreenlet/screens/views/row/WorkflowTaskRowView.kt [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+    private fun getAssetName(workflowTask: WorkflowTask): String {
+        if (workflowTask.comment != null) return workflowTask.comment.type!!
+        if (workflowTask.blogPost != null) return workflowTask.blogPost.type!!
+        return "Undefined"
+    }
+
+    private fun getAssetTitle(workflowTask: WorkflowTask): String {
+        if (workflowTask.comment != null) return workflowTask.comment.text!!
+        if (workflowTask.blogPost != null) return workflowTask.blogPost.headline!!
+        return "Undefined"
```
Comment (victorg1991) [inline android/library/src/main/java/com/liferay/mobile/screens/thingscreenlet/screens/views/row/WorkflowTaskRowView.kt]: what do you think of: `workflowTask.comment?.text ?: workflowTask.blogPost?.headline ?: "Undefined" `

File: android/samples/test-app/src/main/java/com/liferay/mobile/screens/testapp/workflow/activity/WorkflowListActivity.kt [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+        val tasksUrl = resources.getString(R.string.liferay_server) + getTasksUrl()
+
+        workflowScreenlet.load(tasksUrl, credentials = credentials)
+
+        workflowScreenlet.screenletEvents = this
+    }
+
+    private fun getTasksUrl(): String {
+        return "/o/api/p/r/workflow-tasks/assigned-to-me?embedded=comment,blogPost"
+        //return "/o/api/p/r/workflow-tasks/assigned-to-my-roles?embedded=comment,blogPost"
```
Comment (victorg1991) [inline android/samples/test-app/src/main/java/com/liferay/mobile/screens/testapp/workflow/activity/WorkflowListActivity.kt]:
```suggestion
        ''
```

Comment (victorg1991) [inline android/samples/test-app/src/main/java/com/liferay/mobile/screens/testapp/workflow/activity/WorkflowListActivity.kt]: Better not to have commented things in the repo

File: android/library/src/main/java/com/liferay/mobile/screens/thingscreenlet/screens/BaseScreenlet.kt [hunk 4]
```diff
@@ -0,0 +1,13 @@
+package com.liferay.mobile.screens.thingscreenlet.screens
+
+import android.content.Context
+import android.util.AttributeSet
+import android.view.View
+import android.widget.FrameLayout
+
+open class BaseScreenlet @JvmOverloads constructor(
```
Comment (victorg1991) [inline android/library/src/main/java/com/liferay/mobile/screens/thingscreenlet/screens/BaseScreenlet.kt]: Do we need this extracted?

File: android/library/src/main/java/com/liferay/mobile/screens/thingscreenlet/screens/views/row/WorkflowTaskRowView.kt (hunk 2 above)
Comment (dgarciasarai) [reply android/library/src/main/java/com/liferay/mobile/screens/thingscreenlet/screens/views/row/WorkflowTaskRowView.kt]: This is going to change anyway, but it's better your approach. I'll change it, thanks!

File: android/samples/test-app/src/main/java/com/liferay/mobile/screens/testapp/workflow/activity/WorkflowListActivity.kt (hunk 3 above)
Comment (dgarciasarai) [reply android/samples/test-app/src/main/java/com/liferay/mobile/screens/testapp/workflow/activity/WorkflowListActivity.kt]: It's not in the library but I agree :)

File: android/library/src/main/java/com/liferay/mobile/screens/thingscreenlet/screens/BaseScreenlet.kt (hunk 4 above)
Comment (dgarciasarai) [reply android/library/src/main/java/com/liferay/mobile/screens/thingscreenlet/screens/BaseScreenlet.kt]: No, we can keep it inside ThingScreenlet file

**Comment by nhpatt** [comment]
Anything missing?

**Comment by dgarciasarai** [comment]
I don't think so :)

**Comment by dgarciasarai** [comment]
But wait until the code convention PR is merged and then I can format this changes with `detekt`.

**Comment by dgarciasarai** [comment]
Ready to merge :)

**Comment by nhpatt** [comment]
👍

**Pull request merged**